VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room.component.ts

```diff
@@ -0,0 +1,705 @@
+import { Logger } from 'src/app/services/logging/logger-base';
+import { pageUrls } from 'src/app/shared/page-url.constants';
+import { DeviceTypeService } from '../../services/device-type.service';
+import { HearingRole } from '../models/hearing-role-model';
+import { NotificationToastrService } from '../services/notification-toastr.service';
+import { RoomClosingToastrService } from '../services/room-closing-toast.service';
+import { VideoCallService } from '../services/video-call.service';
+import { WaitingRoomBaseDirective } from '../waiting-room-shared/waiting-room-base.component';
+import { TranslateService } from '@ngx-translate/core';
+import { ConsultationInvitationService } from '../services/consultation-invitation.service';
+import { filter, take, takeUntil } from 'rxjs/operators';
+import { UnloadDetectorService } from 'src/app/services/unload-detector.service';
+import { UserMediaService } from 'src/app/services/user-media.service';
+import { ParticipantMediaStatus } from 'src/app/shared/models/participant-media-status';
+import { ConferenceActions } from '../store/actions/conference.actions';
+import { ModalTrapFocus } from '../../shared/modal/modal-trap-focus';
+import { HideComponentsService } from '../services/hide-components.service';
+import { FocusService } from 'src/app/services/focus.service';
+import { ConferenceState } from '../store/reducers/conference.reducer';
+import { Store } from '@ngrx/store';
+import { VHParticipant } from '../store/models/vh-conference';
+import { WaitingRoomUserRole } from './models/waiting-room-user-role';
+import { VideoCallEventsService } from '../services/video-call-events.service';
+import { AudioRecordingService } from 'src/app/services/audio-recording.service';
+import { VideoCallHostActions } from '../store/actions/video-call-host.actions';
+import { CallError } from '../models/video-call-models';
+import { getCountdownComplete } from '../store/selectors/conference.selectors';
+import { VhToastComponent } from 'src/app/shared/toast/vh-toast.component';
+
+@Component({
+    standalone: false,
+    selector: 'app-waiting-room',
+    templateUrl: './waiting-room.component.html',
+    styleUrls: ['./waiting-room.component.scss', '../waiting-room-global-styles.scss']
+})
+export class WaitingRoomComponent extends WaitingRoomBaseDirective implements OnInit, OnDestroy {
+    @Input() userRole: WaitingRoomUserRole;
+    UserRole = WaitingRoomUserRole;
+
+    currentTime: Date;
+
+    showJoinHearingWarning = false;
+    displayLanguageModal: boolean;
+    displayLeaveHearingPopup = false;
+    displayConfirmStartHearingPopup: boolean;
+    displayJoinHearingPopup: boolean;
+    audioErrorRetryToast: VhToastComponent;
+    continueWithNoRecording = false;
+    recordingPaused: boolean;
+    unreadMessageCount = 0;
+
+    private readonly componentLoggerPrefix = '[Waiting Room] -';
+    private destroyedSubject = new Subject();
+
+    constructor(
+        protected route: ActivatedRoute,
```

Comment:
you can get rid of this since the store already has the objects you need.
1) VH Conference is already loaded
2) UserProfile is available, just dispatch the action action to load the profile in the LoggedUserResolveService.

Reply:
it looks like the route is used as a fallback (in waiting-room-base.component.ts) in case the VHConference doesn't have the id:

get conferenceId(): string {
        if (this.vhConference?.id) {
            return this.vhConference.id;
        }
        return this.route.snapshot.paramMap.get('conferenceId');
    }
    
do you think that fallback is no longer needed then?